crates/weaver_forge/src/extensions/otel.rs

```diff
@@ -83,6 +85,72 @@ pub(crate) fn attribute_namespace(input: &str) -> Result<String, minijinja::Erro
     Ok(parts[0].to_owned())
 }
 
+/// Compares two attributes by their requirement_level, then name.
+fn compare_requirement_level(
+    lhs: &Value,
+    rhs: &Value,
+) -> Result<std::cmp::Ordering, minijinja::Error> {
+    fn sort_ordinal_for_requirement(attribute: &Value) -> Result<i32, minijinja::Error> {
+        let level = attribute.get_attr("requirement_level")?;
+        if level
+            .get_attr("conditionally_required")
+            .is_ok_and(|v| !v.is_undefined())
+        {
+            Ok(2)
+        } else if level
+            .get_attr("recommended")
+            .is_ok_and(|v| !v.is_undefined())
+        {
+            Ok(3)
+        } else {
+            match level.as_str() {
+                Some("required") => Ok(1),
+                Some("recommended") => Ok(3),
+                Some("opt_in") => Ok(4),
+                _ => Err(minijinja::Error::custom(format!(
+                    "Expected requirement level, found {}",
+                    level
+                ))),
+            }
+        }
+    }
+    match sort_ordinal_for_requirement(lhs)?.cmp(&sort_ordinal_for_requirement(rhs)?) {
+        std::cmp::Ordering::Equal => {
+            let lhs_name = lhs.get_attr("name")?;
+            let rhs_name = rhs.get_attr("name")?;
+            if lhs_name.lt(&rhs_name) {
+                Ok(std::cmp::Ordering::Less)
+            } else if lhs_name.eq(&rhs_name) {
+                Ok(std::cmp::Ordering::Equal)
+            } else {
+                Ok(std::cmp::Ordering::Greater)
+            }
+        }
+        other => Ok(other),
+    }
+}
+
+/// Sorts a sequence of attributes by their requirement_level, then name.
+pub(crate) fn attribute_sort(input: Value) -> Result<Value, minijinja::Error> {
+    input
+        .as_seq()
+        .map(|values| {
+            let result: Vec<Value> = values
+                .iter()
+                .sorted_by(|lhs, rhs| {
+                    // TODO - Actually output error message here or find another way to do this sorting...
+                    compare_requirement_level(lhs, rhs).unwrap_or(std::cmp::Ordering::Less)
```

Comment:
One way to report errors from this closure is by creating a mutable `Vec` of errors at the top-level scope of the function to collect all errors. Alternatively, replacing the `Vec` with an `Option` can be used to collect only the last observed error.

Reply:
done!